org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandler.java

```diff
@@ -1,194 +1,194 @@
-
-		} // or a build file
-		else if (projectsManager.isBuildFile(file)) {
-			//all errors on that build file should be relevant
-			markers = file.findMarkers(null, true, 1);
-			document = JsonRpcHelpers.toDocument(file);
-		}
-		if (document != null) {
-			this.connection
-					.publishDiagnostics(new PublishDiagnosticsParams(ResourceUtils.toClientUri(uri), toDiagnosticsArray(document, markers)));
-		}
-		return true;
-	}
-
-	/**
-	 * Transforms {@link IMarker}s of a {@link IDocument} into a list of {@link Diagnostic}s.
-	 *
-	 * @param document
-	 * @param markers
-	 * @return a list of {@link Diagnostic}s
-	 */
-	public List<Diagnostic> toDiagnosticsArray(IDocument document, IMarker[] markers) {
-		List<Diagnostic> diagnostics = Stream.of(markers)
-				.map(m -> toDiagnostic(document, m))
-				.filter(d -> d != null)
-				.collect(Collectors.toList());
-		return diagnostics;
-	}
-
-	private Diagnostic toDiagnostic(IDocument document, IMarker marker) {
-		if (marker == null || !marker.exists()) {
-			return null;
-		}
-		Diagnostic d = new Diagnostic();
-		d.setSource(JavaLanguageServerPlugin.SERVER_SOURCE_ID);
-		d.setMessage(marker.getAttribute(IMarker.MESSAGE, ""));
-		d.setCode(String.valueOf(marker.getAttribute(IJavaModelMarker.ID, 0)));
-		d.setSeverity(convertSeverity(marker.getAttribute(IMarker.SEVERITY, -1)));
-		d.setRange(convertRange(document, marker));
-		return d;
-	}
-
-	/**
-	 * @param marker
-	 * @return
-	 */
-	private Range convertRange(IDocument document, IMarker marker) {
-		int line = marker.getAttribute(IMarker.LINE_NUMBER, -1) - 1;
-		int cStart = 0;
-		int cEnd = 0;
-		try {
-			//Buildship doesn't provide markers for gradle files, Maven does
-			if (marker.isSubtypeOf(IMavenConstants.MARKER_ID)) {
-				cStart = marker.getAttribute(IMavenConstants.MARKER_COLUMN_START, -1);
-				cEnd = marker.getAttribute(IMavenConstants.MARKER_COLUMN_END, -1);
-			} else {
-				int lineOffset = 0;
-				try {
-					lineOffset = document.getLineOffset(line);
-				} catch (BadLocationException unlikelyException) {
-					JavaLanguageServerPlugin.logException(unlikelyException.getMessage(), unlikelyException);
-					return new Range(new Position(line, 0), new Position(line, 0));
-				}
-				cEnd = marker.getAttribute(IMarker.CHAR_END, -1) - lineOffset;
-				cStart = marker.getAttribute(IMarker.CHAR_START, -1) - lineOffset;
-			}
-		} catch (CoreException e) {
-			JavaLanguageServerPlugin.logException(e.getMessage(), e);
-		}
-		cStart = Math.max(0, cStart);
-		cEnd = Math.max(0, cEnd);
-
-		return new Range(new Position(line, cStart), new Position(line, cEnd));
-	}
-
-	/**
-	 * @param attribute
-	 * @return
-	 */
-	private DiagnosticSeverity convertSeverity(int severity) {
-		if (severity == IMarker.SEVERITY_ERROR) {
-			return DiagnosticSeverity.Error;
-		}
-		if (severity == IMarker.SEVERITY_WARNING) {
-			return DiagnosticSeverity.Warning;
-		}
-		return DiagnosticSeverity.Information;
-	}
```

Comment:
please avoid huge tabs/whitespace changes, it makes it harder to review what actually changed

Reply:
oops, my local setting is to use space only. will fix it to keep consistent with the repo.